### PR TITLE
feat(issue-search): Add metrics for new snuba search

### PR DESCRIPTION
**Pull request by snigdhas**
Add metrics to track performance for the new GroupAttributes search.

### CODE REVIEW

File: src/sentry/search/snuba/executors.py [hunk 1]
```diff
@@ -1877,5 +1892,10 @@ def query(
 
         groups = group_queryset.in_bulk(paginator_results.results)
         paginator_results.results = [groups[k] for k in paginator_results.results if k in groups]
+        metrics.timing(
```
Comment (JoshFerge) [inline src/sentry/search/snuba/executors.py:1895]: why not just use @metrics.wraps on the whole function?

Comment (snigdhas) [reply src/sentry/search/snuba/executors.py]: smart!

File: src/sentry/search/snuba/executors.py [hunk 2]
```diff
@@ -1660,14 +1661,22 @@ def query(
...
-        if any(sf.key.name in POSTGRES_ONLY_SEARCH_FIELDS for sf in search_filters):
-            max_candidates = options.get("snuba.search.max-pre-snuba-candidates")
-            group_ids_to_pass_to_snuba = list(
-                group_queryset.using_replica().values_list("id", flat=True)[: max_candidates + 1]
-            )
-            if too_many_candidates := (len(group_ids_to_pass_to_snuba) > max_candidates):
-                metrics.incr("snuba.search.too_many_candidates", skip_internal=False)
-                group_ids_to_pass_to_snuba = None
+        max_candidates = options.get("snuba.search.max-pre-snuba-candidates")
+        with sentry_sdk.start_span(op="snuba_group_attributes_query") as span:
```
Comment (JoshFerge) [inline src/sentry/search/snuba/executors.py:1665]: you could also consider doing `@sentry_sdk.tracing.trace` instead this

Comment (JoshFerge) [reply src/sentry/search/snuba/executors.py:1665]: (which is another decorater on the function)

Comment (JoshFerge) [reply src/sentry/search/snuba/executors.py:1665]: oh i guess you want the span to set data on it. this makes sense.

**Review by JoshFerge — APPROVED** [review]

**Pull request merged**